packages/blade/src/components/Table/_decisions/decisions.md

```diff
@@ -0,0 +1,541 @@
+- This would take the relatively more time and effort since internally we will have to build individual components and stitch them together to create the compact API
+- Internally we might not need to put in heavy efforts to ensure composable API is very well structured since we will not be exposing it to the consumers. We would use the composable API internally only to derive the final outcome of a compact API.
+- Strong reasons to go with this approach would be the popularity of the compact API within the Table libraries ecosystem
+- Strong reasons to not go with this approach would be the lack of composability and tree-shaking benefits
+
+##### 3. Both Composable & Compact API
+- We could expose both the Composable & Compact API to the consumers
+- This would take the most amount of implementation time and effort since we will have to build both the APIs
+- This will add a lot of complexity to the library and will make it harder to maintain. Any changes to the Table component, we will have to ensure both APIs are updated accordingly and always maintain parity
+- Swapping out the table library in the future would be challenging since any library we choose we will have to ensure that its feasible to build both kinds of APIs
+- Strong reasons to go with this approach would be to give our consumers complete flexibility to choose the API that they are most comfortable with
+- Strong reasons to not go with this approach would be the complexity and maintainability issues as well as education efforts required to educate the consumers about both the APIs
+
+##### Current API Usage Across Razorpay Projects
+> This is an approximate usage count. We identified this by searching through the codebase. Actual numbers may vary slightly.
+ 
+| Project         | Composable API Table Instances | Compact API Table Instances |
+| --------------- | ------------------------------ | --------------------------- |
+| PG Dashboard    | 101                            | 81                          |
+| Admin Dashboard | 19                             | 196                         |
+| X Dashboard     | 6                              | 11                          |
+
+
+##### Blade Team's Recommendation
+**We recommend** going ahead with the **Compact API** since it gives us the following advantages:
+
+- It is the most popular approach within the Table libraries ecosystem which makes it easier for us to swap out the table library in the future
+- It is easier to consume since it is more concise and requires less effort to create a table
+- TypeScript would be able to help us with type checking the API better since it is a single component
+- Learning curve is lesser since its a single component and not a set of components that need to be composed together
+- As per our findings, 70% of the tables at Razorpay are built using the Compact API which would make it relatively easier for us to migrate them to Blade's Table component as well
+
```

Comment:
Was going through tanstack/table and looks like they also switch to a sort of "composable API" pattern for customizations ([example](https://tanstack.com/table/v8/docs/examples/react/editable-data)), would we need this kind of support as well? Possibly requiring us sometimes to create the table structure manually with table primitives (`th`, `tbody`, etc.) instead of relying on `Table` component.

Sooner than later, usecases will evolve so I like the pattern from tanstack table

Reply:
Tanstack has a very weird API. They're sorta composable and compact at the same time. They seem to have done this to enable framework-agnostic APIs. They do have a massive learning curve though.

Do you forsee any use-cases where we need composable and compact might not work?

Reply:
Mostly for customizability (eg like supporting the above editable data usecase)

Reply:
We could have an `isEditable` & `onTextChange` prop in the columns definition prop for each row which should solve the same with compact API as well, right?

Reply:
Yes, should work for this but for more customizations we'll have to introduce new set of props then

Reply:
That would be true even with composable API... we will have to add new props on other table components instead of the single table component